### DISCUSSION TITLE
Loading a scene with resources (StandardMaterial and Mesh)

### DISCUSSION THREAD

**Discussion by oilandrust** [bevyengine/bevy#7604] [Q&A]
I am trying to use the scene loading facilities of Bevy, I want to build a small in game editor that can save the scenes with DynamicScene::serialize_ron and then load it and spawn a DynamicSceneBundle

I can serialize the scene but the assets are serialized with some id:

"bevy_asset::handle::Handle<bevy_render::mesh::mesh::Mesh>": (

id: Id("8ecbac0f-f545-4473-ad43-e1f4243af51e", 10857144605983285726),

),

And when I load the scene and look at the entities in the inspector it says the handle "points to no asset"

r/bevy - Loading a scene with resources (StandardMaterial and Mesh)
How can I serialize the Mesh along with the scene so that it is found when loading the scene? Do I need to serialize the resources myself and save them to another file that I load before the scene?

Or what is an alternative way to do that?

Any help appreciated.

**Comment by oilandrust** [comment]
I think I need to use gltf meshes because `Mesh` is not serializable.

**Reply by oilandrust** [reply]
I think I might end up saving my own level file with just the information I need to create the level.

**Reply by dimvoly** [reply]
Not sure if this suits your use case, but the programming paradigm seems to be to save the data on the `Component`s, and then create the meshes and the visual stuff from the data after loading those `Component`s. This [crate](https://github.com/Zeenobit/moonshine_save) says something along those lines.
